### DISCUSSION TITLE
How much time it takes to decrypt tally with more than one guardian?

### DISCUSSION THREAD

**Discussion by abbasidd** [microsoft/electionguard#193] [General]
I am working on the electronguard for about two months, everything works fine if I create key ceremony of 1 guardian and 1 quorum. 
However when I create a key ceremony of multiple guardians like quorum of 2 guardian, then it keeps on finding the dicrete log for hours.
Please kindly help me as my job is on stake.

**Comment by abbasidd** [comment]
> I am working on the electronguard for about two months, everything works fine if I create key ceremony of 1 guardian and 1 quorum. 
> However when I create a key ceremony of multiple guardians like quorum of 2 guardian, then it keeps on finding the dicrete log for hours.
> Please kindly help me as my job is on stake. 

And also there was a problem in dlog.py in electionguard. 'Asyncio.run cannot run in the current event loop' and i remove that  asyncio.run. Then it works fine with 1 guardian and quorum of 1.

**Reply by JohnLCaron** [reply]
Hi abbasidd, I saw the same problem when I was doing the Java port. It always indicated a bug in my program, where I was passing an incorrect value to the discrete log. I set the DLOG_MAX value down to 1000 and then threw an Exception when it exceeded, in order to detect that quickly. 

Which is to say it doesnt take much longer to decrypt with more than one guardian.
